### PR TITLE
ft(productDetails): Buyer adds ratings and feedback [#187364831]

### PR DESCRIPTION
**Pull request by Oliviier-dev**
## ℹ️ Description

This PR introduces the feature where the buyers who have purchased a product can be able to leave their feedbacks/reviews on the product; and also anyone who will visit the product page will be able to view all the reviews which were left.

## 🧪 How can this be tested?

This can be found under the product/:productID route

## 📸 Screenshots 
Screenshot 2024-07-12 at 09 18 01
Screenshot 2024-07-12 at 09 18 22
Screenshot 2024-07-12 at 09 18 39

_Please add screenshots of the changes here._

## 📍 Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shebz2023 — DISMISSED** [review]
LGTM

**Review by martinemahirwe — DISMISSED** [review]
LGTM

File: src/components/rootcomponents/ProductDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                    src={product.image_url[2]}
+                    alt={`${product.name} Image 2`}
+                  />
+                </div>
+                <div className="h-fit w-24">
+                  <img
+                    src={product.image_url[3]}
+                    alt={`${product.name} Image 3`}
+                  />
+                </div>
```
Comment (SergeInGithub) [inline src/components/rootcomponents/ProductDetails.tsx]:
```suggestion
  {product.image_url.slice(1, 4).map((url, index) => (
    <div key={index} className="h-fit w-24">
      <img src={url} alt={`${product.name} Image ${index + 1}`} />
    </div>
  ))}
```

File: src/components/rootcomponents/ProductDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                      <RiRefreshLine />
+                    </p>
+                    <div>
+                      <p>Return Delivery</p>
+                      <p>
+                        Free 30 Days Delivery Returns.{' '}
+                        <span className="underline">Details</span>
+                      </p>
+                    </div>
+                  </div>
```
Comment (SergeInGithub) [inline src/components/rootcomponents/ProductDetails.tsx]: This is a component!

File: src/pages/rootpages/AllProducts.tsx [hunk 3]
```diff
@@ -1,12 +1,74 @@
...
+              <label className="inline-flex items-center cursor-pointer">
+                <input
+                  type="radio"
+                  className="form-radio h-4 w-4 text-blue-600"
+                  name="radio"
+                  value="option3"
+                />
+                <span className="ml-2">Fashion Design</span>
+              </label>
+            </div>
```
Comment (SergeInGithub) [inline src/pages/rootpages/AllProducts.tsx]: This is React, Why can't you make this a component? Why are you repeating yourself???

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Work on the reviews provided! @Oliviier-dev

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**